pkg/agent/billing/clients.go

```diff
@@ -108,9 +115,13 @@ func newBlobStorageKeyGenerator(prefix string) func() string {
 		now := time.Now()
 		id := shortuuid.New()
 
-		return fmt.Sprintf("%s/year=%d/month=%02d/day=%02d/%s_%s.ndjson.gz",
+		if prefix != "" {
+			prefix = strings.TrimRight(prefix, "/") + "/"
+		}
+
+		return fmt.Sprintf("%syear=%d/month=%02d/day=%02d/hour=%02d/%s_%s.ndjson.gz",
 			prefix,
-			now.Year(), now.Month(), now.Day(),
+			now.Year(), now.Month(), now.Day(), now.Hour(),
```

Comment:
Same question about preserving backward compatibility.

Reply:
We don't really need it. Nothing depends on it.